docs/tutorials/programmatic-trading-basics.md

```diff
@@ -0,0 +1,304 @@
+---
+title: Programmatic trading basics
+sidebar_position: 2
+hide_title: false
+description: Start bot development for submitting orders with this guide.
+---
+
+In this tutorial you'll learn the basics about how to use Vega and the Vega Wallet to submit orders, so you can build bots or other software to interact with the network. 
+
+This guide covers how to:
+
+1.  [Set up your Vega Wallet](#set-up-your-vega-wallet)
+2.  [Find a market to trade on](#find-a-market-to-trade-on)
+3.  [Get testnet assets using Console](#get-testnet-assets-using-console)
+4.  [Deposit assets](#deposit-assets)
+5.  [Get market information](#get-market-information)
+6.  [Send transactions via the Vega Wallet](#send-transactions-via-the-vega-wallet)
+7.  [Submit an order to place a trade](#submit-an-order-to-place-a-trade)
+8.  [Cancel an order](#cancel-an-order)
+9.  [Monitor trades and overall positions](#monitor-trades-and-overall-positions)
+10. [See example bot code in Go](#example-bot-code-in-go)
+
+This tutorial uses Linux commands. The overall guide will also work for Windows and MacOS, but you may need to update the command format. Each example command with a file path has a `$HOME` variable, which you should replace with your home path.
+
+## Set up your Vega Wallet 
+
+### Download the Vega Wallet software
+
+Each network, such as testnet, will have a compatible version of the CLI Wallet software, which should be used to interact with it. The Vega Wallet software manages wallets and the keys within them, and allows you to approve or reject site connections and transactions.
```

Comment:
I don't think we have anything neatly packaged up yet, but I could create both Go and Java relatively quickly and I think it would be worthwile doing so. I think Jeremy also has a Rust implementation.

Reply:
Here's some Go code for signing transactions and submitting them directly to a core node without the need for a Vega wallet:

https://github.com/MM0819/vega-protocol-auth-go

Should be able to put together the same for a few different languages.

Reply:
Java:

https://github.com/MM0819/vega-protocol-auth-java